types/mongodb/index.d.ts

```diff
@@ -179,7 +179,7 @@ export interface ClientSession extends EventEmitter {
      * @param fn
      * @param options - settings for the transaction
      */
-    withTransaction<T>(fn: WithTransactionCallback<T>, options?: TransactionOptions): Promise<void>;
+    withTransaction(fn: WithTransactionCallback, options?: TransactionOptions): Promise<any>;
```

Comment:
It still returns `void` though:

```suggestion
    withTransaction(fn: WithTransactionCallback, options?: TransactionOptions): Promise<void>;
```

Reply:
It returns void only if it was aborted or if nothing was committed. 
But when something is committed successfully it returns the lump of something which in my test looked like
```
CommandResult {
result: {
  ok: 1,
  '$clusterTime': { clusterTime: [Timestamp], signature: [Object] },
  operationTime: Timestamp { _bsontype: 'Timestamp', low_: 1, high_: 1616019498 }
},
connection: Connection {
```
Any is not the best type, but trying to define the correct `CommandResult` type will be more work than I can justify as most of the time a truthiness check is all that is needed

The example I linked from the mongodb quickstart docs has this example
```
    try {
        const transactionResults = await session.withTransaction(async () => {

            // Some operations here

        }, transactionOptions);

        if (transactionResults) {
            console.log("The reservation was successfully created.");
        } else {
            console.log("The transaction was intentionally aborted.");
        }
    } catch(e){
        console.log("The transaction was aborted due to an unexpected error: " + e);
    } finally {
        await session.endSession();
    }

}
```
With `withTransaction` returning `Promise<void>` typescript errors on the `if (transactionResults) {` line with `An expression of type 'void' cannot be tested for truthiness.`

Reply:
Ahh, I see, maybe `undefined | object` would be a bit more correct then 🤔 

Of course the best thing would be to type `CommandResult`, but in the meantime this is still better than before